nilearn/load_confounds/confounds.py

```diff
@@ -0,0 +1,308 @@
+"""Helper functions for the manipulation of confounds.
+
+Authors: load_confounds team
+"""
+import numpy as np
+import pandas as pd
+from sklearn.decomposition import PCA
+from sklearn.preprocessing import scale
+import os
+import json
+import re
+
+
+img_file_patterns = {
+    "aroma": "_desc-smoothAROMAnonaggr_bold",
+    "nii.gz": "_space-.*_desc-preproc_bold.nii.gz",
+    "dtseries.nii": "_space-.*_bold.dtseries.nii",
+    "func.gii": "_space-.*_hemi-[LR]_bold.func.gii",
+}
+
+img_file_error = {
+    "aroma": (
+        "Input must be ~desc-smoothAROMAnonaggr_bold for full ICA-AROMA"
+        " strategy."
+    ),
+    "nii.gz": "Invalid file type for the selected method.",
+    "dtseries.nii": "Invalid file type for the selected method.",
+    "func.gii": "need fMRIprep output with extension func.gii",
+}
+
+
+def _check_params(confounds_raw, params):
+    """Check that specified parameters can be found in the confounds."""
+    not_found_params = [
+        par for par in params if par not in confounds_raw.columns
+    ]
+    if not_found_params:
+        raise MissingConfound(params=not_found_params)
+    return None
+
+
+def _find_confounds(confounds_raw, keywords):
+    """Find confounds that contain certain keywords."""
+    list_confounds, missing_keys = [], []
+    for key in keywords:
+        key_found = [col for col in confounds_raw.columns if key in col]
+        if key_found:
+            list_confounds.extend(key_found)
+        elif key != "non_steady_state":
+            missing_keys.append(key)
+    if missing_keys:
+        raise MissingConfound(keywords=missing_keys)
+    return list_confounds
+
+
+def _flag_single_gifti(img_files):
```

Comment:
This function is loosely related to confounds, could go to the  _utils module instead.